lld/ELF/SyntheticSections.h

```diff
@@ -591,15 +603,15 @@ class AndroidPackedRelocationSection final : public RelocationBaseSection {
 };
 
 struct RelativeReloc {
-  uint64_t getOffset() const { return inputSec->getVA(offsetInSec); }
+  uint64_t getOffset() const { return inputSec->getVA(reloc->offset); }
 
   const InputSectionBase *inputSec;
-  uint64_t offsetInSec;
+  Relocation *reloc;
```

Comment:
Is adding such indirection applicable here? We need extra info like `sym` and `addend` to be able to convert a relr AUTH reloc to a rela one (see `Writer<ELFT>::finalizeAddressDependentContent()` in e9b53374522fca221414aac3d5a3b225e7fc73d4). If iterating through a vector with `RelativeReloc` items is a hot path, we might potentially encounter a performance decrease here. Is this a known bottle-neck? If yes, what benchmarks (for example, which particular tests from llvm-test-suite) test this?

Reply:
I agree that RelativeReloc needs more information but the current code doesn't work. `&isec.relocs().back()` can will become dangling if the `relocations` variable gets resized.

Reply:
Ah, thanks. Instead of storing pointer to `Relocation` in `RelativeReloc`, changed that to storing its index in the vector. This does not become invalidated when new relocs are appended to the vector and it becomes resized. See b215b0d60ac0c2ac29aeb318da7c95725be6ef86